apps/srt-file-transmit.cpp

```diff
@@ -606,6 +668,110 @@ bool Download(UriParser& srt_source_uri, UriParser& fileuri)
     ExtractPath(path, Ref(directory), Ref(filename));
     Verb() << "Extract path '" << path << "': directory=" << directory << " filename=" << filename;
 
-    return DoDownload(srt_source_uri, directory, filename);
+    return DoDownload(srt_source_uri, directory, filename, cfg, out_stats);
+}
+
+
+int main(int argc, char** argv)
+{
+    FileTransmitConfig cfg;
+    const int parse_ret = parse_args(cfg, argc, argv);
+    if (parse_ret != 0)
+        return parse_ret == 1 ? EXIT_FAILURE : 0;
+
+    //
+    // Set global config variables
+    //
+    if (cfg.chunk_size != SRT_LIVE_MAX_PLSIZE)
+        transmit_chunk_size = cfg.chunk_size;
+    printformat = cfg.stats_pf;
+    transmit_bw_report = cfg.bw_report;
+    transmit_stats_report = cfg.stats_report;
+    transmit_total_stats = cfg.full_stats;
+
+    //
+    // Set SRT log levels and functional areas
+    //
+    srt_setloglevel(cfg.loglevel);
+    for (set<srt_logging::LogFA>::iterator i = cfg.logfas.begin(); i != cfg.logfas.end(); ++i)
+        srt_addlogfa(*i);
+
+    //
+    // SRT log handler
+    //
+    std::ofstream logfile_stream; // leave unused if not set
+    if (!cfg.logfile.empty())
+    {
+        logfile_stream.open(cfg.logfile.c_str());
+        if (!logfile_stream)
+        {
+            cerr << "ERROR: Can't open '" << cfg.logfile.c_str() << "' for writing - fallback to cerr\n";
+        }
+        else
+        {
+            UDT::setlogstream(logfile_stream);
+        }
+    }
+
+    //
+    // SRT stats output
+    //
+    std::ofstream logfile_stats; // leave unused if not set
+    if (cfg.stats_out != "" && cfg.stats_out != "stdout")
+    {
+        logfile_stats.open(cfg.stats_out.c_str());
+        if (!logfile_stats)
+        {
+            cerr << "ERROR: Can't open '" << cfg.stats_out << "' for writing stats. Fallback to stdout.\n";
```

Comment:
Not sure if fallback to stdout is a good idea. Maybe better to simply exit the application with error message.

Reply:
Added return statement.